tests/Backend/ExternalBuckets/SnowflakeExternalBucketShareTest.php

```diff
@@ -180,6 +165,132 @@ public function testExternalSchemaAsSharedBucket(): void
             $result,
         );
 
+        // REFRESH START
+
+        $db->executeQuery(sprintf(
+            'ALTER TABLE %s ADD COLUMN GENDER VARCHAR(50);',
+            SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_TABLE),
+        ));
+
+        $db->executeQuery(sprintf(
+            'UPDATE %s SET GENDER = %s WHERE ID = 1;',
+            SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_TABLE),
+            SnowflakeQuote::quote('male'),
+        ));
+
+        $db->executeQuery(sprintf(
+            'CREATE TABLE %s (ID INT, DESC VARCHAR(255))',
+            SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_TABLE_2),
+        ));
+        $db->executeQuery(sprintf(
+            'INSERT INTO %s (ID, DESC) VALUES (1, %s)',
+            SnowflakeQuote::quoteSingleIdentifier(self::EXTERNAL_TABLE_2),
+            SnowflakeQuote::quote('important description'),
+        ));
+
+        // in test we cannot use GRANT FUTURE, so we have to re-grant all changes to main role
+        // rest re-grants should be done in `refreshBucket`
+        foreach ($guideExploded as $command) {
+            if (str_starts_with($command, 'GRANT') && !str_contains($command, 'FUTURE')) {
+                try {
+                    $db->executeQuery($command);
+                } catch (Exception $e) {
+                    $this->fail($e->getMessage() . ': ' . $command);
+                }
+            }
+        }
```

Comment:
Potom ako prvi krat zaregistrujem bucet tak uz nepotrebujem znaova robit ten guide nie ?

Reply:
Tak, jak máme udělanou tu naši testovací BYODB roli, tak tam nefungují `FUTURE` granty (vysvětloval mi to @jirkasemmler ale už si přesně nepamatuju proč).

Ale v "klasickém" životě fungovat budou.

Takže tady po změnách musím znovu přegrantovat pro projektovou roli nové tabulky (jinak by to udělal `FUTURE GRANT` automaticky), aby ten celý kód fungoval.